File: token-client-core/src/main/java/no/nav/security/token/support/client/core/oauth2/OnBehalfOfGrantRequest.java
package no.nav.security.token.support.client.core.oauth2;

import no.nav.security.token.support.client.core.ClientProperties;
import no.nav.security.token.support.client.core.OAuth2GrantType;

import java.util.Objects;

import static no.nav.security.token.support.client.core.OAuth2GrantType.*;

public class OnBehalfOfGrantRequest extends AbstractOAuth2GrantRequest {
    private final String assertion;

    public OnBehalfOfGrantRequest(ClientProperties clientProperties, String assertion) {
        super(JWT_BEARER, clientProperties);
        this.assertion = assertion;
    }

    String getAssertion() {
        return assertion;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        OnBehalfOfGrantRequest that = (OnBehalfOfGrantRequest) o;
        return Objects.equals(assertion, that.assertion);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), assertion);
    }
}

File: token-client-core/src/main/java/no/nav/security/token/support/client/core/oauth2/OnBehalfOfGrantRequest.kt
package no.nav.security.token.support.client.core.oauth2

import java.util.Objects
import no.nav.security.token.support.client.core.ClientProperties
import no.nav.security.token.support.client.core.OAuth2GrantType
import no.nav.security.token.support.client.core.OAuth2GrantType.Companion.JWT_BEARER

class OnBehalfOfGrantRequest(clientProperties : ClientProperties, val assertion : String) : AbstractOAuth2GrantRequest(JWT_BEARER, clientProperties) {

    override fun equals(other : Any?) : Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        if (!super.equals(other)) return false
        val that = other as OnBehalfOfGrantRequest
        return assertion == that.assertion
    }

    override fun hashCode()  = Objects.hash(super.hashCode(), assertion)
}